### ISSUE TITLE
Reproduce the results

### ISSUE THREAD

**Issue by 0x454447415244** [VChristlein/icdar17code#2]
Hello, I'm trying to reproduce your results. Is it possible to add the missing scripts? (e.g., run_pipeline.py). I have already computed and normalized the SIFT descriptors. I now need to perform dimensionality reduction with PCA. There are many parameters to choose from, that's why I need to make sure that the experiment is the same as in the paper.

Thanks

**Comment by VChristlein** [comment]
Hi, I added the run_pipeline.py (let me know if sth else is missing), a sample config, and a sample call of run_pipeline.py. 
Hope that helps! 

Best,
V.

**Comment by 0x454447415244** [comment]
Thank you Vincent for providing this information. I am trying to run the code but I am not sure what is the format of the input files. Can you please provide sample files, such as the labels files (e.g., icdar17_labels_train.txt) and the features files (ones with _SIFT_patch_pr.pkl.gz). Or at least how they generated. I can generate SIFT features with patches but I'm not sure in what form should I save them in order to use the with your pipeline script.

**Comment by VChristlein** [comment]
I added the icdar17 label files. 

The pkl.gz files are dumps (s. puhma_common.py dump() ) of the extracted features (via the algorithm described below), i.e. a numpy matrix, where each row is one feature and each column represents an activation (penultimate layer of the CNN). 

The features are generated as follows (see also the icdar17 paper): 
- Run SIFT and store features and keypoints (remove duplicates)
- At the keypoints extract 32x32 patches
- Cluster the (training) SIFT descriptors with 5000 clusters using k-means
- Assign cluster labels to train and test patches
- Run a CNN (ResNet20) with the patches as input and the respective cluster indices as target classes (lua code is checked in, but you can of course implement it also in pytorch or similar). 

Let me know if I can help you further!

Best
V.

**Comment by 0x454447415244** [comment]
I checked your paper, it doesn’t mention the accuracy of the CNN on the validation data of the clustered descriptors patches. I know that you perform early stopping. May I know please what is your resnet classification accuracy of the 5k cluster labels on the validation data (the 20k random patches)?
